core/impl/src/main/java/com/blazebit/persistence/impl/EntityMetamodelImpl.java

```diff
@@ -305,7 +305,7 @@ private void discoverEnumTypes(Set<Class<?>> seenTypesForEnumResolving, Map<Stri
             }
 
             discoverEnumTypes(seenTypesForEnumResolving, enumTypes, enumTypesForLiterals, pluralAttribute.getElementType(), baseType, newParent, pluralAttribute, false);
-        } else if (!seenTypesForEnumResolving.contains(fieldType)) {
+        } else if (!seenTypesForEnumResolving.contains(fieldType) && !fieldType.isInterface()) {
```

Comment:
The following would probably be more appropriate, because any attributes have a null `PersistentAttributeType`.

```suggestion
        } else if (!seenTypesForEnumResolving.contains(fieldType) && attribute.getPersistentAttributeType() != null) {
```